examples/step-19/step-19.cc

```diff
@@ -662,21 +666,22 @@ namespace Step19
           for (const auto &particle : particles_in_cell)
             particle_positions.push_back(particle.get_reference_location());
 
-          const Quadrature<dim> quadrature_formula(particle_positions);
-          FEValues<dim>         particle_position_fe_values(mapping,
-                                                    fe,
-                                                    quadrature_formula,
-                                                    update_gradients);
-
-          particle_position_fe_values.reinit(cell);
-
-          // Then we can ask the FEValues object for the gradients of the
-          // solution (i.e., the electric field $\mathbf E$) at these locations
-          // and loop over the individual particles:
-          std::vector<Tensor<1, dim>> field_gradients(
-            quadrature_formula.size());
-          particle_position_fe_values.get_function_gradients(solution,
-                                                             field_gradients);
+          field_values.resize(particle_positions.size());
+          field_gradients.resize(particle_positions.size());
+
+          cell->get_dof_values(solution,
+                               solution_values.begin(),
+                               solution_values.end());
```

Comment:
You should be able to write this as `get_dof_values (solution, solution_values)` if you use `Vector<double>` for `solution_values`.

Reply:
This is still unaddressed.

Reply:
Changed. But the disadvantage is that now we nee to use `make_array_view`.

Reply:
If we are concerned about it we could fix things by adding another constructor to `ArrayView` or, alternatively, let the user pass in a `Vector`. In `DoFAccessor` we are at least consistent with the latter, so it would not feel wrong to do it here as well, even though some uses of `Vector` vs `std::vector` in `FEValues` are confusing in terms of interface consistency.

Reply:
>  adding another constructor to ArrayView

This is not working. Because some static asserts due to:
```
/home/munch/sw-index/dealii/include/deal.II/lac/vector.h:115:5: error: static assertion failed: The Vector class does not support auto-differentiable numbers.
  115 |     !Differentiation::AD::is_ad_number<Number>::value,
```

> , alternatively, let the user pass in a Vector

means that `FEPointEvaluation` would get additional `evaluate`/`integrate` functions!?

Reply:
> If we are concerned about it

And yes, I don't like it ;)

Reply:
Out of interest, where does the instantiation of `AD` numbers come from that the class would complain about?